Tribler/Main/vwxGUI/list_details.py

```diff
@@ -522,20 +525,23 @@ def updateDetailsTab(self):
         self.channel.Show(show_channel)
 
         # Toggle thumbnails
-        # FIXME(lipu): fix the thumbnail path to use metadata
-        thumb_dir = os.path.join(u"",
-                                 binascii.hexlify(self.torrent.infohash))
-        thumb_files = [os.path.join(dp, fn) for dp, _, fns in os.walk(thumb_dir)
-                       for fn in fns if os.path.splitext(fn)[1] in THUMBNAIL_FILETYPES]
-        show_thumbnails = bool(thumb_files)
+        if self.torrent.metadata and 'thumb_hash' in self.torrent.metadata:
```

Comment:
I remember when I was testing using two tribler instances, a injector and a normal tribler, I had this problem that the `modification` messages (the metadata messages) come way slower than the torrents. On the normal tribler instance, I could see the torrents almost immediately after I subscripted to the injector channel, but it took another 20 minutes or more to get those modification messages and show the thumbnails.

It is nice to avoid frequent database I/O here, but I don't know if we do this, the thumbnails will still work as it is now.

I will do some more tests to see if there is any work around.